my_enumerables.rb

```diff
@@ -0,0 +1,203 @@
+      end
+      modified_object
+    else
+      to_enum
+    end
+  end
+
+  def my_all?(pattern = nil)
+    true if length.zero?
+    selected_array = []
+
+    i = 0
+    if block_given?
+      selected_array = my_select { |element| yield element }
+    elsif !block_given? && pattern
+      while i < length
+        selected_array << self[i] if pattern === self[i]
+        i += 1
+      end
+    elsif !block_given? && !pattern
+      while i < length
+        selected_array << self[i] if self[i]
+        i += 1
+      end
+    end
+    selected_array.length == length
+  end
+
+  def my_any?(pattern = nil)
+    false if length.zero?
+    selected_array = []
+
+    i = 0
+    if block_given?
+      selected_array = my_select { |element| yield element }
+    elsif !block_given? && pattern
+      while i < length
+        selected_array << self[i] if pattern === self[i]
+        i += 1
+      end
+    elsif !block_given? && !pattern
+      while i < length
+        selected_array << self[i] if self[i]
+        i += 1
+      end
+    end
+    selected_array.length.positive?
+  end
+
+  def my_none?(pattern = nil)
+    true if length.zero?
+    selected_array = []
+
+    i = 0
+    if block_given?
+      selected_array = my_select { |element| yield element }
+    elsif !block_given? && pattern
+      while i < length
+        selected_array << self[i] if pattern === self[i]
+        i += 1
+      end
+    elsif !block_given? && !pattern
+      while i < length
+        selected_array << self[i] if self[i]
+        i += 1
+      end
+    end
+    selected_array.length.zero?
+  end
+
+  def my_count(*args)
+    count_array = []
+
+    i = 0
+    if block_given?
+      while i < length
+        count_array << self[i] if yield self[i]
+        i += 1
+      end
+    elsif !block_given? && !args.empty?
+      while i < length
+        count_array << self[i] if self[i] == args.first
+        i += 1
+      end
+    else
+      count_array = self
+    end
+    count_array.length
+  end
+
+  def my_map
+    object = to_a
+    modified_object = []
+
+    i = 0
+    if block_given?
+      while i < to_a.length
+        modified_object << (yield object[i])
+        i += 1
+      end
+      modified_object
+    else
+      to_enum
+    end
+  end
+
+  def my_inject(initial = nil, method = nil)
+    modified_object = []
+    if block_given?
+      if initial
+        memo = initial
+        modified_object = my_map { |element| memo = yield memo, element }
+      else
+        memo = first
+        modified_object = to_a[1..-1].my_map { |element| memo = yield memo, element }
+      end
+    elsif !block_given?
+      if initial && method
+        memo = initial
+        modified_object = my_map { |element| memo = memo.send(method, element) }
+      elsif initial
+        method = initial
+        memo = first
+        modified_object = to_a[1..-1].my_map { |element| memo = memo.send(method, element) }
+      end
+    end
+    modified_object[-1]
+  end
```

Comment:
my_inject` identical and returns the same thing as ruby's [`inject`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-inject). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_inject.png) from [Odin](https://github.com/TheOdinProject/curriculum/blob/master/ruby_programming/archive/basic_ruby/project_advanced_building_blocks.md#project-2-enumerable-methods))
- raises a "LocalJumpError" when no block or argument is given. But you one does not cover any case similar to this issue.